src/docs/platform-architecture-reference/platform-automation.md

```diff
@@ -0,0 +1,81 @@
+
+keywords: resource, reclamation, tool, automation, security
+
+page_purpose: Describes the automation we run on to manage the OpenShift platform.
+
+audience: technical lead, developer
+
+author: Matt Spencer, Steven Barre
+
+content_owner: Olena Mitovska
+
+sort_order: 6
+---
+
+# Platform Automation
+
+We use some automation on the platform to help manage resource usage, alert about misconfigured objects and to encourage teams to use images that are secure and up to date. 
+
+To date, resource utilization (the use of CPU and RAM) on our platforms has not been efficient. For example, almost 50% of CPU capacity on the Silver cluster is reserved by apps, but not actually used. This unused utilization increases costs and impacts our ability to onboard new teams.  Sometimes teams do not maintain and update the images that their deployments are based on. This can lead to security vulnerabilities, incompatibilities and other problems  
+ 
+To address this, we’ve developed automated tools that monitor apps on the platform. 
+ 
+These tools looks for these types of objects: 
+- `Deployments` and `deploymentconfigs` that are based on images have not changed in over a year
+- `Deployments`, `deploymentconfigs` and `statefulsets` that are crashing constantly 
+
+Emails are sent to the Product Owner and Technical Leads registered for the Product in the [Registry](https://registry.developer.gov.bc.ca/). If you want to change who gets the emails, go update the contacts in the Registry.
+
+## Deployments and deploymentconfigs that have not changed in a year 
+
+This tool checks the `lastUpdateTime` of the `Progressing` block in the `status` field of the Deployment or DeploymentConfig. If it is close to a year old, but not over, a warning email will be sent. If the deployment is managed by ArgoCD then a warning email is sent asking for action to be taken. This is because ArgoCD will just scale it back up after the tool scaled it down. If not managed by ArgoCD and the timestamp is over a year then the deployment is scaled to zero replicas and an email is sent.
+
+You can check the timestamp on your deployment in the YAML under `status`.
+ 
+ ```yaml
+   conditions:
+  - lastTransitionTime: "2021-11-25T17:04:01Z"
+    lastUpdateTime: "2021-11-25T17:04:11Z"
+    message: replication controller "backup-postgresql-1" successfully rolled out
+    reason: NewReplicationControllerAvailable
+    status: "True"
+    type: Progressing
+  ```
+## Deployments, deploymentconfigs and statefulsets that are crashing constantly
+
+This tool looks at the restart count of pods. If a pod has more than 100 restarts it looks for the controlling object of the pod, be it aDeployment, DeploymentConfig, or StatefulSet. If the object is managed by ArgoCD then a warning email is sent asking for action to be taken. This is because ArgoCD will just scale it back up after the tool scaled it down. If not managed by ArgoCD, the object is scaled to zero replicas and an email is sent.
```

Comment:
```suggestion
This tool looks at the restart count of pods. If a pod has more than 100 restarts it looks for the controlling object of the pod, be it a Deployment, DeploymentConfig, or StatefulSet. If the object is managed by ArgoCD then a warning email is sent asking for action to be taken. This is because ArgoCD will just scale it back up after the tool scaled it down. If not managed by ArgoCD, the object is scaled to zero replicas and an email is sent.
```